### PR TITLE
RChain: store categories in one variable

### LINKED ISSUE #671: rChain: categories
- [x] in rChain inject only one variable called ```categories```
- in ```server/modules/analysis/service/rChainService/rFile/system``` add listCategories as model object to init the scripts
  - [x] fetch categories used by calculations
  - [x] create list categories with all fetched categories ```categories <- list(c1, c2...)``` 
  - [x] update names of categories with category names
  - [x] remove categories 
  - [x] append the content after fetching all data
- [x] in persist-results, use join with listCategories object

**Example of read-data**
```
category_name1 <- arena.get(...)
category_name2 <- arena.get(...)
categories <- list(category_name1, category_name2)
names(categories) <- c('category_name1', 'category_name2')
rm(category_name1)
rm(category_name2)
```

**Example of join in persist-results**
```
entityResults $var_name_TMP <- treeResults$var_name
entityResults $var_name <- NULL
category <- categories$category_name1
entityResults <- sqldf('
            SELECT 
                r.*,
                "{itemUuid: """ || c.uuid || """, code: """ || c.code || """, label: """ || c.label || """}" AS var_name
            FROM entityResults r
            LEFT OUTER JOIN category c
            ON r.var_name_TMP = c.code  
        ')
entityResults$var_name_TMP <- NULL
rm(category)
```

### PR DESCRIPTION
**Pull request by SteRiccio**
Resolves #671

### CODE REVIEW

File: server/modules/analysis/service/rChainService/rChain.js [hunk 1]
```diff
@@ -144,6 +151,10 @@ class RChain {
     this._fileLogin = await new RFileLogin(this).init()
     this._fileReadData = await new RFileReadData(this).init()
 
+    // Init categories
+    this._listCategories = new ListCategories(this)
+    await this._fileReadData.appendContent(...this._listCategories.scripts)
```
Comment (minotogna) [inline server/modules/analysis/service/rChainService/rChain.js:156]: Add this to ```RFileReadData.init()```

Comment (SteRiccio) [reply server/modules/analysis/service/rChainService/rChain.js:156]: what about DFResults? Should it access listCategories with rChain.fileReadData.listCategories then? Or should RChain expose listCategories from the instance variable _fileReadData?

File: server/modules/analysis/service/rChainService/rChain.js [hunk 2]
```diff
@@ -162,7 +170,7 @@ class RChain {
   }
 
   async init() {
-    await this._initSurveyAndChain()
+    await this._initSurveyChainAndCategories()
```
Comment (minotogna) [inline server/modules/analysis/service/rChainService/rChain.js:173]: I would leave the old name, categories are part of the survey, plus you are initializing them here

Comment (SteRiccio) [reply server/modules/analysis/service/rChainService/rChain.js:173]: ok

**Pull request merged**